### DISCUSSION TITLE
Change font type in grid axis (not in coordinate system)

### DISCUSSION THREAD

**Discussion by kaathaarinaa** [pyvista/pyvista#3905] [Q&A] [answered]
Hello :).

I want to change the font size, type, etc... (for using the plot in a publication). I can successfully change the size, but not the font, neither of the grid axis label nor of the tick label. (MWE below)

As to what I read, there are at least 3 options of changing font and font size: 

1. Changing the local or global plotting themes: https://docs.pyvista.org/examples/02-plot/themes.html 
2. Overwriting the global default values: https://docs.pyvista.org/api/plotting/_autosummary/pyvista.themes.DarkTheme.font.html --> ```pyvista.global_theme.font.family = 'arial'``` (font) or ```pyvista.global_theme.font.size= 20``` (font size)
3. Chaning the font size is possible directly when adding the grid: ```pl.show_grid(..., font_size=20)``` Such an option is not available for the font itself though

Option 1. or 2. seem most appealing to me but I tried them all and none of them let me change the font (changing font size works with all of them). The font stays the same irrespective of whether I use "times", "courier" or "arial". I think it is always "arial".

Below is a MWE, where, if you change from ```pyvista.global_theme.font.family = 'arial'``` to ```pyvista.global_theme.font.family = 'times'```, nothing happens. I would very much aprreciate any hints!

```
import pyvista
from pyvista import examples
from pyvista import themes


# ----- Option 2 (not used) -----
# pyvista.global_theme.font.family = 'arial'
# pyvista.global_theme.font.size = 20


# ----- Option 1 (used) -----
def set_theme():
    my_theme = themes.DocumentTheme()
    my_theme.font.family = 'arial'  # should change font but does not...
    my_theme.font.size = '10'  # changes axis label & tick label size
    return my_theme


# Get dummy data
mesh = examples.load_uniform()
# Set plot theme style according to Option 1
my_theme = set_theme()
pyvista.set_plot_theme(my_theme)
# Plot
pl = pyvista.Plotter()
actor = pl.add_mesh(mesh, color='red')
pl.show_grid(color='k', xlabel='x', ylabel='y', zlabel='z')  # , font_size=20)  # Option 3 (not used)
pl.show()
 ```

**Comment by akaszynski** [comment, marked as answer]
This is indeed an issue and is being resolved in https://github.com/pyvista/pyvista/pull/3910.

Expect this resolved in `pyvista==0.38.0`, which should be out by the end of the week. For now you could check out the branch and install from there.

Also, consider seeing #3897 for improving screenshot quality. That will be out in next release as well.

**Reply by banesullivan** [reply]
> which should be out by the end of the week.

I'm setting a hard deadline of Wednesday morning 😄

**Reply by kaathaarinaa** [reply]
setting font works after switching to #3910.

Regarding the image quality: I tried #3897 and the scaling works as suggested, but I find it more convenient to use ```plotter.save_graphics()``` and setting the windows size large initially and tinkering with the font size to get an appropriate resolution for documents/publications.

As I mentioned in #3914, it would be a huge update for publication readiness of pyvista plots if gridlines/axis were also vectorised. Currently (on #3910), font/text is vectorised, but grilines/axis are not. Surface/contour is not either but this is intended.
